pallets/grants/src/lib.rs

```diff
@@ -375,9 +344,16 @@ impl<T: Config> Pallet<T> {
 			.checked_add(&schedule_amount)
 			.ok_or(Error::<T>::NumOverflow)?;
 
-		T::Currency::transfer(from, to, schedule_amount, ExistenceRequirement::AllowDeath)?;
-		T::Currency::set_lock(VESTING_LOCK_ID, to, total_amount, WithdrawReasons::all());
-		<VestingSchedules<T>>::mutate(to, |v| (*v).push(schedule));
+		<VestingSchedules<T>>::try_mutate(to, |vesting_schedules| -> DispatchResult {
+			vesting_schedules
+				.try_push(schedule)
+				.map_err(|_| <Error<T>>::MaxScheduleOverflow)?;
```

Comment:
How about moving this to line 343 before calculating total_amount so we don't calculate total_amount if we there is vesting schedule vector is going to overflow.

Reply:
Is it possible to check with a  `schedule.len()` which should be a quick operation?

Reply:
@simonsso If we could check the `schedule.len()` and then do `push` rather than `try_push`, then your suggestion could be possibly better. However on a bounded vector you can only do `try_push` meaning we still need to cover the error case here anyway.